spec/index.bs

```diff
@@ -6767,19 +6789,62 @@ interface GPUCanvasContext {
         given device.
 
         <div algorithm="GPUCanvasContext.getSwapChainPreferredFormat">
-            **Called on:** {{GPUCanvasContext}} this.
+            **Called on:** {{GPUCanvasContext}} |this|.
 
             **Arguments:**
             <pre class=argumentdef for="GPUCanvasContext/getSwapChainPreferredFormat(device)">
-                device: Device the swap chain format should be queried for.
+                |device|: Device the swap chain format should be queried for.
             </pre>
 
             **Returns:** {{Promise}}&lt;{{GPUTextureFormat}}&gt;
 
-            Issue: Describe {{GPUCanvasContext/getSwapChainPreferredFormat()}} algorithm steps.
+            1. Let |promise| be [=a new promise=].
```

Comment:
The more I think about it, the more it seems like this could be synchronous: either the canvas is HDR and the preferred format is rgba16float or rgb10a2unorm depending on how it was configured, or it is not HDR in which case it is best as bgra8unorm on all OSes except Android where it is best rgba8unorm. Basically we know synchronously on canvas context creation what's the best format.

Reply:
I think we need to put some more thought into this (with input from platform/canvas experts) because, like I mentioned above, I'm not exactly sure how browsers are going to handle this. Why do we need the canvas - is it because we can check which display the canvas is on? What happens if the canvas moves between displays, does the result here change? Do we instead provide the "max" of all the attached displays? What happens if there's a non-HDR display and an HDR display and you unplug the HDR display?